diff --git a/node_modules/react-native-elements/src/buttons/ButtonGroup.js b/node_modules/react-native-elements/src/buttons/ButtonGroup.js
index fc9c897..30a056a 100644
--- a/node_modules/react-native-elements/src/buttons/ButtonGroup.js
+++ b/node_modules/react-native-elements/src/buttons/ButtonGroup.js
@@ -197,7 +197,7 @@ const styles = {
     backgroundColor: 'transparent',
   },
   disabledText: theme => ({
-    color: color(theme.colors.disabled).darken(0.3),
+    color: color(theme.colors.disabled).darken(0.3).toString(),
   }),
   disabledSelected: theme => ({
     backgroundColor: theme.colors.disabled,
diff --git a/node_modules/react-native-elements/src/index.d.ts b/node_modules/react-native-elements/src/index.d.ts
index ff9ff85..8aa4354 100644
--- a/node_modules/react-native-elements/src/index.d.ts
+++ b/node_modules/react-native-elements/src/index.d.ts
@@ -830,7 +830,7 @@ export interface InputProps extends TextInputProperties {
   /**
    * Renders component in place of the React Native `TextInput` (optional)
    */
-  inputComponent?: React.ComponentClass<any>;
+  inputComponent?: React.ComponentType<any>;
 
   /**
    * 	Adds styling to input component (optional)
